website/client/components/auth/registerLogin.vue

```diff
@@ -0,0 +1,230 @@
+<template lang="pug">
+.form-wrapper
+  #top-background
+    .seamless_stars_varied_opacity_repeat
+
+  #login-form
+    .text-center
+      div
+        .svg-icon.gryphon(v-html="icons.gryphon")
+      div
+        .svg-icon.habitica-logo(v-html="icons.habiticaIcon")
+    .form-group.row.text-center
+      .col-6
+        .btn.btn-secondary.social-button(@click='socialAuth("facebook")', v-once)
+          .svg-icon.social-icon(v-html="icons.facebookIcon")
+          | {{this.registering ? $t('signUpWithSocial', {social: 'Facebook'}) : $t('loginWithSocial', {social: 'Facebook'})}}
+      .col-6
+        .btn.btn-secondary.social-button(@click='socialAuth("google")', v-once)
+          .svg-icon.social-icon(v-html="icons.googleIcon")
+          | {{this.registering ? $t('signUpWithSocial', {social: 'Google'}) : $t('loginWithSocial', {social: 'Google'})}}
+    .form-group
+      label(for='usernameInput', v-once) {{$t('username')}}
+      input#usernameInput.form-control(type='text', :placeholder='$t("usernamePlaceholder")', v-model='username')
+    .form-group(v-if='registering')
+      label(for='emailInput', v-once) {{$t('email')}}
+      input#emailInput.form-control(type='email', :placeholder='$t("emailPlaceholder")', v-model='email')
+    .form-group
+      label(for='passwordInput', v-once) {{$t('password')}}
+      input#passwordInput.form-control(type='password', :placeholder='$t("passwordPlaceholder")', v-model='password')
+    .form-group(v-if='registering')
+      label(for='confirmPasswordInput', v-once) {{$t('confirmPassword')}}
+      input#confirmPasswordInput.form-control(type='password', :placeholder='$t("confirmPasswordPlaceholder")', v-model='passwordConfirm')
+      small.form-text(v-once) {{$t('termsAndAgreement')}}
+    .text-center
+      .btn.btn-info(@click='register()', v-if='registering', v-once) {{$t('joinHabitica')}}
+      .btn.btn-info(@click='login()', v-if='!registering', v-once) {{$t('login')}}
+
+  #bottom-background
+    .seamless_mountains_demo_repeat
+    .midground_foreground_extended2
+</template>
+
+<style lang="scss" scoped>
+  @import '~client/assets/scss/colors.scss';
+
+  .form-wrapper {
+    background-color: $purple-200;
+  }
+
+  #login-form {
+    margin: 0 auto;
+    width: 40em;
+    padding-top: 5em;
+    padding-bottom: 22.5em;
+    position: relative;
+    z-index: 1;
+
+    .gryphon {
+      width: 63.2px;
+      height: 69.4px;
+    }
+
+    .habitica-logo {
+      width: 144px;
+      height: 31px;
+      margin-top: 2em;
+      margin-bottom: 2em;
+    }
+
+    label {
+      color: $white;
+      font-weight: bold;
+    }
+
+    input {
+      margin-bottom: 2em;
+      border-radius: 2px;
+      background-color: #432874;
+      border-color: transparent;
+      height: 50px;
+      color: $purple-400;
+
+      ::-webkit-input-placeholder { /* Chrome/Opera/Safari */
```

Comment:
these should not be necessary as sass adds browser specific prefixes automatically